fuzz.go

```diff
@@ -521,13 +536,18 @@ var unicodeRanges = []charRange{
 }
 
 // randString makes a random string up to 20 characters long. The returned string
-// may include a variety of (valid) UTF-8 encodings.
-func randString(r *rand.Rand) string {
+// may include a variety of (valid) UTF-8 encodings. If wants specific encoding
+// string, it should input character range.
+func randString(r *rand.Rand, characterRange *charRange) string {
 	n := r.Intn(20)
 	sb := strings.Builder{}
 	sb.Grow(n)
 	for i := 0; i < n; i++ {
-		sb.WriteRune(unicodeRanges[r.Intn(len(unicodeRanges))].choose(r))
+		if characterRange == nil {
```

Comment:
(If I hadn't made my other suggestion, I'd be pointing out here that it's much better to make both callers pass in a list of ranges and not make this function have an exception where it uses global data.)

Reply:
For using function `CustomStringFuzzFunc` of `UnicodeRange` and `UnicodeRanges`, the logic using unicodeRange is changed.

I add defaultUnicodeRanges in package, for those who just want to generate string as usual.